cumulusci/tasks/github/tests/test_publish.py

```diff
@@ -0,0 +1,366 @@
+                    "email": "testuser@testdomain.com",
+                }
+            ),
+        )
+
+    @responses.activate
+    @mock.patch("cumulusci.tasks.github.publish.download_extract_github_from_repo")
+    @mock.patch("cumulusci.tasks.github.publish.CommitDir")
+    def test_run_task(self, commit_dir, extract_github):
+        responses.add(
+            method=responses.GET,
+            url=self.repo_api_url,
+            json=self._get_expected_repo(owner=self.repo_owner, name=self.repo_name),
+        )
+        responses.add(
+            method=responses.GET,
+            url=self.repo_api_url,
+            json=self._get_expected_repo(
+                owner=self.public_owner, name=self.public_name
+            ),
+        )
+        responses.add(
+            responses.GET,
+            self.repo_api_url + "/releases/latest",
+            json=self._get_expected_release("release/1.0"),
+        )
+        responses.add(
+            method=responses.GET,
+            url=self.repo_api_url + "/git/refs/tags/release/1.0",
+            status=201,
+            json=self._get_expected_tag_ref("release/1.0", "SHA"),
+        )
+        responses.add(
+            method=responses.GET,
+            url=self.repo_api_url + "/git/tags/SHA",
+            json=self._get_expected_tag("release/1.0", "SHA"),
+            status=201,
+        )
+        responses.add(
+            responses.GET,
+            self.repo_api_url + "/releases/tags/release/1.0",
+            json=self._get_expected_release("release/1.0"),
+        )
+        responses.add(
+            responses.GET,
+            self.public_repo_url + "/releases/tags/release/1.0",
+            status=404,
+        )
+        responses.add(
+            responses.GET,
+            self.public_repo_url + "/git/refs/tags/release/1.0",
+            status=404,
+        )
+        responses.add(
+            responses.POST,
+            self.public_repo_url + "/releases",
+            json=self._get_expected_release("release"),
+        )
+        responses.add(
+            responses.POST,
+            self.repo_api_url + "/git/tags",
+            json=self._get_expected_tag("release/1.0", "SHA0"),
+            status=201,
+        )
+        responses.add(
+            responses.POST, self.repo_api_url + "/git/refs", json={}, status=201
+        )
+        task_config = TaskConfig(
+            {
+                "options": {
+                    "branch": "master",
+                    "version": "latest",
+                    "repo_url": self.public_repo_url,
+                    "includes": ["tasks/foo.py", "unpackaged/pre/foo/package.xml"],
+                }
+            }
+        )
+        extract_github.return_value.namelist.return_value = [
+            "tasks/foo.py",
+            "unpackaged/pre/foo/package.xml",
+            "force-app",
+        ]
+        task = PublishSubtree(self.project_config, task_config)
+        task()
```

Comment:
Can this be more explicit about asserting something, like the fact that the expected requests were actually made?